### PR TITLE
Add package resolution to resolve namespace / files to class name and vice versa

### PR DESCRIPTION
**Pull request by sandrokeil**

### CODE REVIEW

File: src/Package/Psr4Info.php [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+    private function normalizeNamespace(string $namespace): string
+    {
+        $namespace = \str_replace('\\\\', '\\', $namespace);
+        $splitNamespace = \explode('\\', $namespace);
+
+        return \trim(\implode('\\', $splitNamespace), '\\');
+    }
+
+    /**
+     * PSR-4 folders must be upper camel case
```
Comment (codeliner) [inline src/Package/Psr4Info.php:163]: Is this true according to the PSR-4 standard?

File: src/Package/Psr4Info.php [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+     *
+     * @param string $path
+     * @return string
+     */
+    private function normalizePath(string $path): string
+    {
+        $path = \preg_replace('/^' . \addcslashes($this->sourceFolder, '/') . '\//', '', $path);
+        $splitPath = \explode('/', $path);
```
Comment (codeliner) [inline src/Package/Psr4Info.php:171]: Why explode and implode without any logic in between?

File: src/Package/Psr4Info.php [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+    /**
+     * Removes duplicates of backslashes and trims backslashes
+     *
+     * @param string $namespace
+     * @return string
+     */
+    private function normalizeNamespace(string $namespace): string
+    {
+        $namespace = \str_replace('\\\\', '\\', $namespace);
+        $splitNamespace = \explode('\\', $namespace);
```
Comment (codeliner) [inline src/Package/Psr4Info.php:157]: Why explode and implode without any logic in between?

**Review by codeliner — CHANGES_REQUESTED** [review]

File: src/Package/Psr4Info.php (hunk 1 above)
Comment (sandrokeil) [reply src/Package/Psr4Info.php:163]: No, it's only case sensitive.

File: src/Package/Psr4Info.php (hunk 2 above)
Comment (sandrokeil) [reply src/Package/Psr4Info.php]: Good catch

File: src/Package/Psr4Info.php (hunk 3 above)
Comment (sandrokeil) [reply src/Package/Psr4Info.php]: Good catch

**Review by codeliner — APPROVED** [review]

**Pull request merged**